e2e-testing/__tests__/orders.test.ts

```diff
@@ -0,0 +1,387 @@
+  Order_Side,
+  Order_TimeInForce,
+  OrderFlags, SocketClient,
+  SubaccountInfo,
+  ValidatorClient,
+  HeightResponse,
+} from '@dydxprotocol/v4-client-js';
+import {
+  DYDX_LOCAL_ADDRESS,
+  DYDX_LOCAL_ADDRESS_2,
+  DYDX_LOCAL_MNEMONIC,
+  DYDX_LOCAL_MNEMONIC_2,
+} from './helpers/constants';
+import { DateTime } from 'luxon';
+import * as utils from './helpers/utils';
+import {
+  FillTable,
+  FillType,
+  Liquidity,
+  OrderSide,
+  OrderTable,
+  SubaccountTable,
+  helpers,
+  IsoString,
+  CandleResolution,
+} from '@dydxprotocol-indexer/postgres';
+
+const PERPETUAL_PAIR_BTC_USD: number = 0;
+const quantums: Long = new Long(1_000_000_000);
+const subticks: Long = new Long(1_000_000_000);
+
+const defaultOrder: IPlaceOrder = {
+  clientId: 0,
+  orderFlags: OrderFlags.SHORT_TERM,
+  clobPairId: PERPETUAL_PAIR_BTC_USD,
+  side: Order_Side.SIDE_BUY,
+  quantums,
+  subticks,
+  timeInForce: Order_TimeInForce.TIME_IN_FORCE_UNSPECIFIED,
+  reduceOnly: false,
+  clientMetadata: 0,
+};
+
+type OrderDetails = {
+  mnemonic: string;
+  timeInForce: number;
+  orderFlags: number;
+  side: number;
+  clobPairId: number;
+  quantums: number;
+  subticks: number;
+};
+
+const orderDetails: OrderDetails[] = [
+  {
+    mnemonic: DYDX_LOCAL_MNEMONIC,
+    timeInForce: 0,
+    orderFlags: 64,
+    side: 1,
+    clobPairId: PERPETUAL_PAIR_BTC_USD,
+    quantums: 10000000,
+    subticks: 5000000000,
+  },
+  {
+    mnemonic: DYDX_LOCAL_MNEMONIC_2,
+    timeInForce: 0,
+    orderFlags: 64,
+    side: 2,
+    clobPairId: PERPETUAL_PAIR_BTC_USD,
+    quantums: 5000000,
+    subticks: 5000000000,
+  },
+];
+
+async function placeOrder(
+  mnemonic: string,
+  order: IPlaceOrder,
+): Promise<void> {
+  const wallet = await LocalWallet.fromMnemonic(mnemonic, BECH32_PREFIX);
+  const client = await ValidatorClient.connect(Network.local().validatorConfig);
+
+  const subaccount = new SubaccountInfo(wallet, 0);
+  const modifiedOrder: IPlaceOrder = order;
+  if (order.orderFlags !== 0) {
+    // cancel the order 30 seconds from now
+    modifiedOrder.goodTilBlock = 0;
+    const now = new Date();
+    const millisecondsPerSecond = 1000;
+    const interval = 30 * millisecondsPerSecond;
+    const future = new Date(now.valueOf() + interval);
+    modifiedOrder.goodTilBlockTime = Math.round(future.getTime() / 1000);
+  } else {
+    modifiedOrder.goodTilBlockTime = 0;
+  }
+
+  await client.post.placeOrderObject(
+    subaccount,
+    modifiedOrder,
+  );
+}
+
+describe('orders', () => {
+  it('test orders', async () => {
+    const indexerClient = new IndexerClient(Network.local().indexerConfig);
+    const heightResp: HeightResponse = await indexerClient.utility.getHeight();
+    const height: number = heightResp.height;
+    connectAndValidateSocketClient();
+
+    // place all orders
+    for (const order of orderDetails) {
+      const modifiedOrder: IPlaceOrder = defaultOrder;
+      modifiedOrder.clientId = Math.floor(Math.random() * 1000000000);
```

Comment:
nit: move into a helper so it's obvious what this is doing

Reply:
done